rustls/src/crypto/mod.rs

```diff
@@ -186,6 +206,63 @@ pub struct CryptoProvider {
     pub key_provider: &'static dyn KeyProvider,
 }
 
+impl CryptoProvider {
+    /// Sets the supplied `CryptoProvider` as the default for this process.
+    ///
+    /// This can be called successfully at most once in any process execution.
+    ///
+    /// Call this early in your process to configure which provider is used for
+    /// the provider.  The configuration should happen before any use of
+    /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
+    pub fn install_as_process_default(provider: impl Into<Arc<Self>>) -> Result<(), Arc<Self>> {
+        PROCESS_DEFAULT_PROVIDER.set(provider.into())
+    }
+
+    /// Sets the default provider to the one named unambiguously by rustls crate features.
+    ///
+    /// This function does nothing if the crate features are ambiguous (ie, specify two
+    /// providers), or specify no providers.  In both cases the application should
+    /// explicitly specify the provider to use with [`CryptoProvider::install_as_process_default`].
+    pub fn install_process_default_from_crate_features() {
+        #[cfg(all(feature = "ring", not(feature = "aws_lc_rs")))]
+        {
+            // Ignore the error resulting from us losing a race, and accept the outcome.
+            let _ = Self::install_as_process_default(ring::default_provider());
+        }
+
+        #[cfg(all(feature = "aws_lc_rs", not(feature = "ring")))]
+        {
+            let _ = Self::install_as_process_default(aws_lc_rs::default_provider());
+        }
+    }
+
+    /// Returns the default `CryptoProvider` for this process.
+    ///
+    /// This will be `None` if no default has been set yet.
+    pub fn process_default() -> Option<Arc<Self>> {
```

Comment:
IMO it would be nicer to yield `&Arc<Self>` here. `cloned()` is cheap but not free.